gst/tensor_ros_sink/tensor_ros_sink.c

```diff
@@ -0,0 +1,849 @@
+
+  return TRUE;
+}
+
+/**
+ * @brief Setter for value last_render_time.
+ */
+static void
+gst_tensor_ros_sink_set_last_render_time (GstTensorRosSink * self,
+    GstClockTime now)
+{
+  g_return_if_fail (GST_IS_TENSOR_ROS_SINK (self));
+
+  g_mutex_lock (&self->mutex);
+  self->last_render_time = now;
+  g_mutex_unlock (&self->mutex);
+}
+
+/**
+ * @brief Getter for value last_render_time.
+ */
+static GstClockTime
+gst_tensor_ros_sink_get_last_render_time (GstTensorRosSink * self)
+{
+  GstClockTime last_render_time;
+
+  g_return_val_if_fail (GST_IS_TENSOR_ROS_SINK (self), GST_CLOCK_TIME_NONE);
+
+  g_mutex_lock (&self->mutex);
+  last_render_time = self->last_render_time;
+  g_mutex_unlock (&self->mutex);
+
+  return last_render_time;
+}
+
+/**
+ * @brief Setter for value signal_rate.
+ */
+static void
+gst_tensor_ros_sink_set_signal_rate (GstTensorRosSink * self, guint rate)
+{
+  g_return_if_fail (GST_IS_TENSOR_ROS_SINK (self));
+
+  silent_debug ("set signal_rate to %d", rate);
+  g_mutex_lock (&self->mutex);
+  self->signal_rate = rate;
+  g_mutex_unlock (&self->mutex);
+}
+
+/**
+ * @brief Getter for value signal_rate.
+ */
+static guint
+gst_tensor_ros_sink_get_signal_rate (GstTensorRosSink * self)
+{
+  guint rate;
+
+  g_return_val_if_fail (GST_IS_TENSOR_ROS_SINK (self), 0);
+
+  g_mutex_lock (&self->mutex);
+  rate = self->signal_rate;
+  g_mutex_unlock (&self->mutex);
+
+  return rate;
+}
+
+/**
+ * @brief Setter for flag emit_signal.
+ */
+static void
+gst_tensor_ros_sink_set_emit_signal (GstTensorRosSink * self, gboolean emit)
+{
+  g_return_if_fail (GST_IS_TENSOR_ROS_SINK (self));
+
+  silent_debug ("set emit_signal to %d", emit);
+  g_mutex_lock (&self->mutex);
+  self->emit_signal = emit;
+  g_mutex_unlock (&self->mutex);
+}
+
+/**
+ * @brief Getter for flag emit_signal.
+ */
+static gboolean
+gst_tensor_ros_sink_get_emit_signal (GstTensorRosSink * self)
+{
+  gboolean res;
+
+  g_return_val_if_fail (GST_IS_TENSOR_ROS_SINK (self), FALSE);
+
+  g_mutex_lock (&self->mutex);
+  res = self->emit_signal;
+  g_mutex_unlock (&self->mutex);
+
+  return res;
+}
+
+/**
+ * @brief Setter for flag silent.
+ */
+static void
+gst_tensor_ros_sink_set_silent (GstTensorRosSink * self, gboolean silent)
+{
+  g_return_if_fail (GST_IS_TENSOR_ROS_SINK (self));
+
+  silent_debug ("set silent to %d", silent);
+  self->silent = silent;
+}
+
+/**
+ * @brief Getter for flag silent.
+ */
+static gboolean
+gst_tensor_ros_sink_get_silent (GstTensorRosSink * self)
+{
+  g_return_val_if_fail (GST_IS_TENSOR_ROS_SINK (self), TRUE);
+
+  return self->silent;
+}
+
+/**
+ * @brief Function to initialize the plugin.
+ *
+ * See GstPluginInitFunc() for more details.
+ */
+NNSTREAMER_PLUGIN_INIT (tensor_ros_sink)
+{
+  GST_DEBUG_CATEGORY_INIT (gst_tensor_ros_sink_debug, "tensor_ros_sink",
+      0, "tensor_ros_sink element");
+
+  return gst_element_register (plugin, "tensor_ros_sink",
+      GST_RANK_NONE, GST_TYPE_TENSOR_ROS_SINK);
+}
+
+#ifndef SINGLE_BINARY
```

Comment:
These lines are unnecessary now (PR #868 in nnstreamer, SINGLE_BINARY removed)

Reply:
@jaeyun-jung Removed.